File: Backend/TrackrAPI/TrackrAPI/Dtos/GestionExpediente/ExpedienteTratamientoDetalleDto.cs
using TrackrAPI.Models;

namespace TrackrAPI.Dtos.GestionExpediente
{
    public class ExpedienteTratamientoDetalleDto
    {
        public int IdExpedienteTratamiento { get; set; }
        public int IdExpediente { get; set; }
        public string Farmaco { get; set; } = null!;
        public DateTime FechaRegistro { get; set; }
        public DateTime FechaInicio { get; set; }
        public DateTime? FechaFin { get; set; }
        public decimal Cantidad { get; set; }
        public string Unidad { get; set; }
        public string Indicaciones { get; set; }
        public string Padecimiento { get; set; }
        public int IdPadecimiento { get; set; }
        public int IdUsuarioDoctor { get; set; }
        public string NombreDoctor { get; set; }
        public string ApellidosDoctor { get; set; }
        public string? TituloDoctor { get; set; }
        public string? ImagenBase64 { get; set; }
        public string? TipoMime { get; set; }
        public bool RecordatorioActivo { get; set; }
        public bool[] DiaSemana { get; set; }
        public TimeSpan[] Horas { get; set; }
        public DateTime[] Bitacora { get; set; }
    }
}

File: Frontend/trackr-paciente/src/app/shared/Dtos/gestion-perfil/expediente-tratamiento-detalle-dto.ts
export class ExpedienteTratamientoDetalleDto{
    idExpedienteTratamiento?: number;
    idExpediente?: number;
    farmaco : string;
    // fechaRegistro : Date;
    fechaInicio?: Date;
    fechaFin?: Date;
    cantidad: number
    unidad : string;
    indicaciones : string;
    padecimiento? : string;
    idPadecimiento : number;
    idUsuarioDoctor: number
    nombreDoctor?: string;
    apellidosDoctor?: string;
    tituloDoctor?: string;
    archivo: any;
    archivoTipoMime:string='';
    archivoNombre:string='';
    imagenBase64: string;
    tipoMime? : string;
    recordatorioActivo: boolean; // Recordatorio de tomas
    diaSemana: boolean[]; // Checkboxes dias de la semana
    horas: string[]; // Horario(h)
    bitacora?: Date[]; // Bitacora Consumo de Medicamentos
}
